prov/efa/src/rdm/rxr_ep.c

```diff
@@ -1905,9 +1905,11 @@ static inline void rdm_ep_poll_ibv_cq_ex(struct rxr_ep *ep, size_t cqe_to_proces
 			rxr_pkt_handle_rma_completion(ep, pkt_entry);
 			break;
 		case IBV_WC_RECV_RDMA_WITH_IMM:
+			pkt_entry = (void *)(uintptr_t)ep->ibv_cq_ex->wr_id;
```

Comment:
Why is this same line repeated 5 times, why not just set it at the top of the loop?

Reply:
I added a second commit to address this refactor.